src/index.tsx

```diff
@@ -11,117 +11,160 @@ import Animated, {
+        if (textHeight.value === 0 || parentHeight.value === 0) {
           return 0;
         }
-        return Math.round(parentWidth.value / textWidth.value) + 1;
-      },
-      (v) => {
-        if (v === 0) {
-          return;
+        return Math.round(parentHeight.value / textHeight.value) + 1;
+      } else {
+        // Horizontal
+        if (textHeight.value === 0 || parentHeight.value === 0) {
+          return 0;
         }
+        return Math.round(parentHeight.value / textHeight.value) + 1;
+      }
+    },
+    (v) => {
+      if (v === 0) {
+        return;
+      }
+
+      if (direction === 'up' || direction === 'down') {
+        runOnJS(setCloneTimes)(v);
+      } else {
         // This is going to cover the case when the text/element size
         // is greater than the actual parent size
         // Double this to cover the entire screen twice, in this way we can
         // reset the position of the first element when its going to move out
         // of the screen without any noticible glitch
         runOnJS(setCloneTimes)(v * 2);
-      },
-      []
-    );
-    return (
-      <Animated.View
-        style={style}
-        onLayout={(ev) => {
-          parentWidth.value = ev.nativeEvent.layout.width;
-        }}
-        pointerEvents="box-none"
-      >
-        <Animated.View style={styles.row} pointerEvents="box-none">
-          {
-            // We are adding the text inside a ScrollView because in this way we
-            // ensure that its not going to "wrap".
-          }
-          <Animated.ScrollView
-            horizontal
-            style={styles.hidden}
-            pointerEvents="box-none"
-          >
-            <View
-              onLayout={(ev) => {
-                textWidth.value = ev.nativeEvent.layout.width;
-              }}
-            >
-              {children}
-            </View>
-          </Animated.ScrollView>
-          {cloneTimes > 0 &&
-            [...Array(cloneTimes).keys()].map((index) => {
-              return (
-                <AnimatedChild
-                  key={`clone-${index}`}
-                  index={index}
-                  anim={anim}
-                  textWidth={textWidth}
-                  spacing={spacing}
-                >
-                  {children}
-                </AnimatedChild>
-              );
-            })}
-        </Animated.View>
+      }
+    },
+    [],
+  );
+  return (
+    <Animated.View
+      style={style}
+      onLayout={(ev) => {
+        parentHeight.value = ev.nativeEvent.layout.height;
+        parentWidth.value = ev.nativeEvent.layout.width;
+      }}
+      pointerEvents="box-none">
+      <Animated.View style={styles.row} pointerEvents="box-none">
+        {
+          // We are adding the text inside a ScrollView because in this way we
+          // ensure that its not going to "wrap".
+        }
+        <Animated.ScrollView style={styles.hidden} pointerEvents="box-none">
```

Comment:
no horizontal support for more than one Text

I think it's possible to do it like this:
horizontal={direction === 'left' || direction === 'right'}